Library/Homebrew/sorbet/tapioca/utils.rb

```diff
@@ -0,0 +1,23 @@
+# typed: strict
+# frozen_string_literal: true
+
+module Homebrew
+  module Tapioca
+    module Utils
+      # @param class_methods [Boolean] whether to get class methods or instance methods
+      # @return the `module` methods that are defined in the given file
+      sig {
+        params(mod: Module, file_name: String,
+               class_methods: T::Boolean).returns(T::Array[T.any(Method, UnboundMethod)])
+      }
+      def self.methods_from_file(mod, file_name, class_methods: false)
+        methods = if class_methods
+          mod.methods(false).map { mod.method(_1) }
+        else
+          mod.instance_methods(false).map { mod.instance_method(_1) }
+        end
+        methods.select { _1.source_location&.first&.end_with?(file_name) }
```

Comment:
This is pretty fragile (I added warning comments to `Attrable` above (and I think that parlour was better in this specific regard), but there is prior art in tapioca [itself](https://github.com/Shopify/tapioca/blob/3948457/lib/tapioca/dsl/compilers/graphql_input_object.rb#L67) for using source location to filter methods. I'm open to other suggestions.